core/src/test/java/org/jboss/pnc/rex/test/GenericHttpClientStandaloneTest.java

```diff
@@ -0,0 +1,198 @@
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Slf4j
+@QuarkusTest
+@TestSecurity(authorizationEnabled = false)
+public class GenericHttpClientStandaloneTest {
+
+    private WireMockServer wireMockServer;
+    private static final int MOCK_SERVER_PORT = 8083;
+
+    @BeforeEach
+    void setup() {
+        wireMockServer = new WireMockServer(WireMockConfiguration.options().port(MOCK_SERVER_PORT));
+        wireMockServer.start();
+        WireMock.configureFor("localhost", MOCK_SERVER_PORT);
+    }
+
+    @AfterEach
+    void teardown() {
+        if (wireMockServer != null) {
+            wireMockServer.stop();
+        }
+    }
+
+    @Inject
+    GenericVertxHttpClient httpClient;
+
+    @ParameterizedTest
+    @ValueSource(ints = {425, 429, 500, 503, 599})
+    void shouldRetryOnErrorCodeAndSucceed(int code) throws InterruptedException {
+        // given
+        // 1st request: respond with error
+        stubFor(get(urlPathMatching("/.*"))
+                    .willReturn(aResponse().withStatus(code))
+                    .inScenario("retry-" + code)
+                    .whenScenarioStateIs(Scenario.STARTED)
+                    .willSetStateTo("attempt1"));
+        // 2nd request: respond with error
+        stubFor(get(urlPathMatching("/.*"))
+                    .willReturn(aResponse().withStatus(code))
+                    .inScenario("retry-" + code)
+                    .whenScenarioStateIs("attempt1")
+                    .willSetStateTo("attempt2"));
+        // 3rd request: respond with success
+        stubFor(get(urlPathMatching("/.*"))
+                    .willReturn(aResponse().withStatus(200))
+                    .inScenario("retry-" + code)
+                    .whenScenarioStateIs("attempt2")
+                    .willSetStateTo("attempt3"));
+
+        ArrayBlockingQueue<HttpResponse<Buffer>> responses = new ArrayBlockingQueue<>(10);
+        Consumer<HttpResponse<Buffer>> onResponse = r -> {
+            responses.add(r);
+        };
+
+        // when
+        httpClient.makeReactiveRequest(
+            URI.create("http://localhost:" + MOCK_SERVER_PORT),
+            Method.GET,
+            Collections.emptyList(),
+            "",
+            onResponse,
+            (r) -> {})
+            .await().atMost(Duration.of(5, ChronoUnit.SECONDS));
+
+        // expect
+        HttpResponse<Buffer> response = responses.poll(5, TimeUnit.SECONDS);
+        assertThat(response.statusCode()).isEqualTo(200);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {425, 429, 500, 503, 599})
```

Comment:
I guess don't try with 425 cause there's other Test that requires 5s timeout and you shouldn't be able to configure 'max-retries' together with 'timeout'. If that's possible, then it's a bug. 

Optionally, if you want to keep 425 here, you could create a new profile like in `WithoutTaskCleaning.class` and change the config before tests run.

Reply:
prolly not relevant anymore